### ISSUE TITLE
AssertionError: One of self.n_classes or n_classes must be an integer

### ISSUE THREAD

**Issue by Shyamapada34** [Project-MONAI/tutorials#496]
Hi Monai Team,
First of all, thank you for providing the good repository for image processing work. I was trying to implement UNETR algorithm given which is given in "monai/tutorials/3d_segmentation/unetr_btcv_segmentation_3d.ipynb". I have taken dataset same as given notbook, like training = 80, validation = 20, but when I was running the it, getting an error : 'AssertionError: One of self.n_classes or n_classes must be an integer'
Please let me know, how solve this issue. I have changes data different, not resolved. Please find the attached screenshot. 
![Error](https://user-images.githubusercontent.com/74256866/147755692-936aa92d-626e-4d0f-aea1-c770b468dfd1.JPG)
 
Thanks and Regards
Shyama

**Comment by Nic-Ma** [comment]
Hi @Shyamapada34 ,

Thanks for your interest and experiments.
Could you please help provide your MONAI version?
The `AsDiscrete` args changed in v0.8.0, so I think you can try again with the latest MONAI.

Thanks.

**Comment by Shyamapada34** [comment]
Hi Nic-Ma,
Thanks for your quick response. I have changed the MONAI version and the problem is solved.
The UNETR model is trained with my own dataset. Now, I want to test the model with another set of nifty data and want to save the predicted image in nifty format with original spacing scale. In the training phase, I have resampling the images accordingly: (1.5,1.5,0.8).  
Please provide the steps. 
Happy New Year !


Thanks and Regards
Shyama

**Comment by Nic-Ma** [comment]
Hi @Shyamapada34 ,

Cool, glad to see your progress.
Please refer to this inference tutorial to invert the prediction image and save:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/torch/unet_inference_dict.py#L70

Thanks.

**Comment by ahatamiz** [comment]
Hi @Shyamapada34,

In addition to great suggestions by @Nic-Ma , I would like to mention that we also provide the option of on-the-fly resampling for both train and testing the UNETR repostiory:
https://github.com/ahatamiz/research-contributions/blob/master/UNETR/BTCV/test.py#L40

For instance, you may pass `--space_x=1.5, --space_y=1.5, --space_x=0.8` for training on (1.5,1.5,0.8) resolution. 

Thanks

**Comment by Shyamapada34** [comment]
Hi Ahatamizadeh,
Thank you so much for reply. I am working with it. It is very helpful to implement. 

Thanks and Regards
Shyama

**Comment by Shyamapada34** [comment]
Hi Ahatamizadeh,
Thanks for your reply. Your comments are very helpfull. I have trained the UNETR model for pancreas data with output channel 3, such as background, pancreas and cancer, and tested the model with new data set. To visualize the predicted image, I was opened the predicted image on ImageJ tool, it was not a single 3D image, it was 3 sets of 3D image. I also checked with also Mango tool, same problem happed. I want a single 3D image where segmented background, pancreas and cancer will be there. 

Thanks and Regards
Shyama
